### PR TITLE
Models kz

### PR DESCRIPTION
**Pull request by kennethchen91**
### Assignee Tasks

* Defined Models
* [x] Products
* [x] Orders
* [x] OrderLineItems
* [x] OrderStatus
* [x] ShippingMethods
* [x] Categories

### Guidelines

Defined database tables.

---

Test PR Notes

### CODE REVIEW

File: server/db/models/Category.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const Sequelize = require('sequelize');
+const db = require('../db');
+
+const Category = db.define('category', {
+  name: {
+    type: Sequelize.STRING,
+    allowNull: false
```
Comment (adamvare) [inline server/db/models/Category.js]: Do you need a validate: {notEmpty: true}?

File: server/db/models/Order.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+const Order = db.define('order', {
+  grandTotal: {
+    type: Sequelize.FLOAT,
+    allowNull: false
```
Comment (adamvare) [inline server/db/models/Order.js]: validate: {notEmpty: true} probably needed?

File: server/db/models/Category.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  }
+});
+
+Category.beforeValidate(category => {
+  /*
+   * Generate slug
+   */
+  if (!category.slug) {
+    category.slug = category.name
+      .replace(/\s/g, '_')
```
Comment (adamvare) [inline server/db/models/Category.js]: dashes -, Not underscores _

File: server/db/models/product.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+const Sequelize = require('sequelize')
+const db = require('../db')
+
+const Product = db.define('product', {
+  name: {
+    type: Sequelize.STRING,
+    allowNull: false
```
Comment (adamvare) [inline server/db/models/product.js]: validate: {notEmpty: true}

File: server/db/models/product.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  }
+})
+
+Product.beforeValidate(product => {
+  /*
+   * Generate slug
+   */
+  if (!product.slug) {
+    product.slug = product.name
+      .replace(/\s/g, '_')
```
Comment (adamvare) [inline server/db/models/product.js]: dashes - instead of _

File: server/db/models/Category.js (hunk 1 above)
Comment (kennethchen91) [reply server/db/models/Category.js]: done

File: server/db/models/Order.js (hunk 2 above)
Comment (kennethchen91) [reply server/db/models/Order.js]: What if the order hasn't been purchased yet and is still in cart?

**Pull request merged**